assets/react/controllers/Incident/List.jsx

```diff
@@ -0,0 +1,234 @@
+import { Table, Tag, Avatar, Row, Col, Badge, Switch } from "antd";
+import TaskContext from "./TaskContext";
+import _ from "lodash";
+import "antd/lib/pagination/style/index.css";
+import "antd/lib/tree/style/index.css";
+import { useTranslation } from "react-i18next";
+
+async function _fetchIncidents() {
+  const httpClient = new window._auth.httpClient();
+  return await httpClient.get(
+    window.Routing.generate("api_incidents_get_collection"),
+  );
+}
+
+function _showPriority(priority) {
+  switch (priority) {
+    case 1:
+      return { text: "HIGH", status: "error" };
+    case 2:
+      return { text: "MEDIUM", status: "warning" };
+    case 3:
+      return { text: "LOW", status: "success" };
+  }
+}
+
+function _statusCtx({ store, order: { restaurant } }) {
+  if (store.id) {
+    return (
+      <div>
+        <i className="fa fa-truck mr-2" aria-hidden="true"></i>
+        {store.name}
+      </div>
+    );
+  }
+
+  if (restaurant.id) {
+    return (
+      <div>
+        <i className="fa fa-cutlery mr-2" aria-hidden="true"></i>
+        {restaurant.name}
+      </div>
+    );
+  }
+}
+
+function _storeFilter(value, record) {
+  if (value.startsWith("#s")) {
+    return record.store.id === parseInt(value.substring(2));
+  }
+
+  if (value.startsWith("#r")) {
+    return record.order.restaurant.id === parseInt(value.substring(2));
+  }
+
+  if (value === "store") {
+    return record.store.id;
+  }
+
+  if (value === "restaurant") {
+    return record.order.restaurant.id;
+  }
+}
+
+export default function () {
+  const { t } = useTranslation();
+  const [incidents, setIncidents] = useState(null);
+  const [showClosed, setShowClosed] = useState(false);
+
+  const users = useMemo(() => {
+    return _(incidents)
+      .uniqBy("author.id")
+      .map((u) => ({ value: u.author.username, text: u.author.username }))
+      .value();
+  }, [incidents]);
+
+  const stores = useMemo(() => {
+    return _(incidents)
+      .filter((i) => i.store.id)
+      .uniqBy("store.id")
+      .map((i) => ({ value: "#s" + i.store.id, text: i.store.name }))
+      .value();
+  }, [incidents]);
+
+  const restaurants = useMemo(() => {
+    return _(incidents)
+      .filter((i) => i.order.restaurant.id)
+      .uniqBy("order.restaurant.id")
+      .map((i) => ({
+        value: "#r" + i.order.restaurant.id,
+        text: i.order.restaurant.name,
+      }))
+      .value();
+  }, [incidents]);
+
+  const customers = useMemo(() => {
+    return _(incidents)
+      .filter((i) => i.order.customer.id)
+      .uniqBy("order.customer.id")
+      .map((i) => ({
+        value: i.order.customer.id,
+        text: i.order.customer.username,
+      }))
+      .value();
+  }, [incidents]);
+
+  const showedIncidents = useMemo(() => {
+    if (showClosed) {
+      return incidents;
+    }
+    return _(incidents)
+      .filter((i) => i.status !== "CLOSED")
+      .value();
+  }, [incidents, showClosed]);
+
```

Comment:
it looks like the `incidents` should be saved in redux, and all these `useMemo` should become redux selectors (maybe at some point later 😄)

Reply:
For now this screen fit in one component, so there is no need to use a store. (but yeah maybe later (: )